CMakeLists.txt

```diff
@@ -253,9 +253,11 @@ rocm_package_setup_component(tests
 )
 
 add_subdirectory(library)
-add_subdirectory(example)
-add_subdirectory(test)
-add_subdirectory(profiler)
+if(BUILD_ALL)
+    add_subdirectory(example)
+    add_subdirectory(test)
+    add_subdirectory(profiler)
+endif()
```

Comment:
If we're going to make conditional build of particular CK code parts I'd suggest to go one step ahead and make it possible to build more fine-grained. 
First of all, I'd suggest to use [CMake's options](https://cmake.org/cmake/help/latest/command/option.html) like the one for [`BitInt` extension](https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/CMakeLists.txt#L24). This will make it clear if the particular flag is set on or off. 
Second I'd suggest to add additional flags like `WITH_EXAMPLES`, `WITH_TESTS`, `WITH_PROFILER` which will control compilation of respective CK's code parts. The above flags I think would be especially useful for developers who no longer would need to manually comment out `add_subdirectory(...)` lines to build only code parts that they need.

Reply:
OK, so how about something like this:

option(BUILD_ALL, "Whether to build all targets, such as examples, tests, and profiler." OFF)
option(WITH_EXAMPLES, "Whether to build examples." OFF)
option(WITH_TESTS, "Whether to build tests." OFF)
option(WITH_PROFILER, "Whether to build profiler." OFF)
if(BUILD_ALL)
    add_subdirectory(example)
    add_subdirectory(test)
    add_subdirectory(profiler)
else()
   if(WITH_EXAMPLES)
        add_subdirectory(example)
   endif()
   if(WITH_TESTS)
       add_subdirectory(test)
   endif()
   if(WITH_PROFILER)
      add_subdirectory(profiler)
    endif()
endif()

So we could either build all targets with a single flag, or select one of each targets with one individual flag.